src/main/java/seedu/address/model/EventTime.java

```diff
@@ -51,6 +60,21 @@ public static EventTime parse(String startTime, String endTime) throws DateTimeP
         return new EventTime(LocalTime.parse(first, COMPACT_TIME_FORMAT), LocalTime.parse(second, COMPACT_TIME_FORMAT));
     }
 
+    /**
+     * Gets EventTime Object representation of duration of delivery task.
+     *
+     * @param duration startTime - endTime.
+     * @return the duration with the specified start and end time
+     */
+    public static EventTime parse(String duration) throws DateTimeParseException {
+        //split string into 3 parts to get start time, "-" and end time
+        String[] durationArr = duration.split(" ");
+        String startTimeStr = durationArr[0];
+        String endTimeStr = durationArr[2];
+
+        return parse(startTimeStr, endTimeStr);
```

Comment:
Okay noted, because usually we check if is valid then we parse, so i didn't include it.

But for safety, i will include isValidDuration under the parse and create a IncorrectFormat exception for it to throw

Reply:
adding assert instead

Reply:
adding assert instead